engine/src/main/java/org/terasology/engine/subsystem/ThreadManagerSubsystem.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2015 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.engine.subsystem;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.terasology.assets.module.ModuleAwareAssetTypeManager;
+import org.terasology.config.Config;
+import org.terasology.context.Context;
+import org.terasology.engine.ComponentSystemManager;
+import org.terasology.engine.modes.GameState;
+import org.terasology.monitoring.ThreadActivity;
+import org.terasology.monitoring.ThreadMonitor;
+import org.terasology.utilities.concurrency.ShutdownTask;
+import org.terasology.utilities.concurrency.Task;
+import org.terasology.utilities.concurrency.TaskMaster;
+
+import java.util.concurrent.RejectedExecutionException;
+
+/**
+ * @author Immortius
+ */
```

Comment:
I wonder if this class or the method submitTask() could benefit from a javadoc specifying a little what happens to tasks that have been submitted. I.e. can any assumption be made on when the task will run or guarantees of the order in which they will run? I can also see that they are run with minimum priority. That notion perhaps could also go in the javadoc. 

Finally, could perhaps the number of threads be made a public variable (still static/final though) for any interested party? Somebody might attempt to submit 3417 task at once and not quite understand why they are not started all at once. Obviously.